src/plugins/navigation/public/top_nav_menu/top_nav_menu_item.tsx

```diff
@@ -39,24 +40,29 @@ export function TopNavMenuItem(props: TopNavMenuData) {
     props.run(e.currentTarget);
   }
 
-  const btn = (
-    <EuiButtonEmpty
-      size="xs"
-      isDisabled={isDisabled()}
-      onClick={handleClick}
-      data-test-subj={props.testId}
-      className={props.className}
-    >
+  const commonButtonProps = {
+    isDisabled: isDisabled(),
+    onClick: handleClick,
+    iconType: props.iconType,
+    iconSide: (props.iconRight ? 'right' : 'left') as 'right' | 'left' | undefined,
+    'data-test-subj': props.testId,
+  };
+
+  const btn = props.emphasize ? (
+    <EuiButton {...commonButtonProps} size="s" fill style={{ fontSize: 'smaller' }}>
```

Comment:
I would ask @elastic/kibana-design to add `xs` support on `EuiButton`, but it's not urgent.